### DISCUSSION TITLE
Schema and "def _resource_and_event(request)"

### DISCUSSION THREAD

**Discussion by rust-jvm-go** [webhookdb/webhookdb#854] [Q&A] [answered]
I'm following https://docs.webhookdb.com/guides/build-replicator.

For Brevo https://developers.brevo.com/reference/getemaileventreport-1:

The backfill API endpoint (https://api.brevo.com/v3/smtp/statistics/events) returns a JSON array of `events`, thus:

```
{
    "events": [
        {
            "email": "a@b.com",
            "date": "2024-01-23T09:36:46.364+08:00",
            "subject": "Brevo Test Transactional Mail",
            "messageId": "<202401230136.35720869946@smtp-relay.mailin.fr>",
            "event": "requests",
            "tag": "",
            "ip": "xx.xx.xx.xx",
            "from": "c@d.com"
        },
       ...
    ]
}
```

However, the webhook only returns a single event.

https://developers.brevo.com/docs/how-to-use-webhooks

https://developers.brevo.com/docs/transactional-webhooks

My questions are:
1. How to define the schema for named JSON arrays?
2. Does the `_resource_and_event()` method get invoked for both webhooks and backfills, such that there must be handling for both?

**Comment by rgalanakis** [comment, marked as answer]
See example in this PR (glad you figured it out): https://github.com/webhookdb/webhookdb/pull/859
